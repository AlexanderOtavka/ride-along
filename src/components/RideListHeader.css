/**
 * RideListHeader.css
 *
 * Created by Zander Otavka on 6/5/17.
 * Copyright (C) 2016  Grinnell AppDev.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

.header {
  position: relative;
}

.headerBottom, .headerTop {
  display: flex;
  padding: 16px;
}

.headerTop {
  flex-direction: row;
}

.headerBottom {
  position: absolute;
  right: 0;
  bottom: 0;
  left: 0;
  flex-direction: column;
  padding-top: 32px;
  z-index: -1;
}

.headerBottom.inSearchMode {
  transform: translateY(100%);
}

.header:global(.request) .headerTop {
  background-color: #F2C94C;
}

.header:global(.offer) .headerTop {
  background-color: #219653;
}

.header:global(.request) .headerBottom {
  background-color: #E2B93C;
}

.header:global(.offer) .headerBottom {
  background-color: #118663;
}

.modeSwitch {
  display: flex;
  flex-direction: row;
  margin-left: 16px;
}

.downChevron {
  position: absolute;
  top: -2px; /* because svgs have a weird gap */
  right: 0;
  left: 0;
  width: 100%;
}

.header:global(.request) .downChevron {
  fill: #F2C94C;
}

.header:global(.offer) .downChevron {
  fill: #219653;
}
