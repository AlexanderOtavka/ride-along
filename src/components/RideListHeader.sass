/**
 * RideListHeader.sass
 *
 * Created by Zander Otavka on 6/5/17.
 * Copyright (C) 2016  Grinnell AppDev.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

@import "style/colors"
@import "style/layout"
@import "style/animations"

.header
  position: relative
  z-index: 1

.headerBottom, .headerTop
  display: flex
  padding: 16px

.headerTop
  position: relative
  flex-direction: row
  align-items: center
  z-index: 1
  overflow: hidden
  transition: background $touch-effect-timing

.downChevron
  position: absolute
  top: -8px // because svgs have a weird gap
  right: 0
  left: 0
  width: 100%

.headerBottom
  position: absolute
  right: 0
  bottom: 0
  left: 0
  flex-direction: column
  padding-top: 32px
  z-index: 0
  transition: transform $header-slide-timing

  &.inSearchMode
    transform: translateY($header-slide-distance)

.modeSwitch
  display: flex
  flex-direction: row
  margin-left: 16px

.closeButton
  margin-right: 10px
  margin-left: 26px

.closeIcon
  fill: $off-white !important
  width: 28px !important

.closeRipple
  background-color: $off-white

.header.request
  .headerTop
    background-color: $primary-yellow

  .downChevron
    fill: $primary-yellow

  .headerBottom
    background-color: darken($primary-yellow, 10%)

.header.offer
  .headerTop
    background-color: $primary-green

  .downChevron
    fill: $primary-green

  .headerBottom
    background-color: darken($primary-green, 10%)
