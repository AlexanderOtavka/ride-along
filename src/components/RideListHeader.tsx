/**
 * @file RideListHeader.tsx
 *
 * Created by Zander Otavka on .
 * Copyright (C) 2016  Grinnell AppDev.
 *
 * @license
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import React from "react"
import { Link } from "react-router-dom"
import { Form, RadioGroup } from "react-form"
import { IconButton } from "react-toolbox/lib/button"
import classnames from "classnames"

import BoxField from "./BoxField"
import ModeButton from "./ModeButton"

import { RideSearchModel } from "../store/rides"

import * as routes from "../constants/routes"

import styles from "./RideListHeader.sass"

import DownChevronSVG from "../drawables/down-chevron.svg"
import CloseSVG from "../drawables/close.svg"
import CurrentLocationSVG from "../drawables/crosshairs-gps.svg"

export interface Props {
  isSearchMode: boolean
  values: RideSearchModel
  onSearchModeChange: (isSearchMode: boolean, values: RideSearchModel) => void
  onValuesChange: (values: RideSearchModel) => void
}

function RideListHeader({ isSearchMode, ...props }: Props) {
  return (
    <header className={classnames(styles.header, styles[props.values.mode])}>
      <Form
        values={isSearchMode ? props.values : { mode: props.values.mode }}
        onChange={({ values }: any) => props.onValuesChange(values)}
        component={false}
      >
        {({ submitForm, values }: any) =>
          <form
            onSubmit={ev => {
              ev.preventDefault()

              if (!isSearchMode) {
                props.onSearchModeChange(true, values)
              }

              submitForm()
            }}
            action={routes.rides.search}
            method="get"
          >
            <div className={styles.headerTop}>
              <BoxField
                field="departLocation"
                type={isSearchMode ? "text" : "submit"}
                placeholder={
                  isSearchMode
                    ? "Departure location"
                    : values.mode === "offer"
                      ? "Offer a ride"
                      : "Request a ride"
                }
                autoFocus={
                  isSearchMode &&
                  values.departLocation === undefined &&
                  values.arriveLocation === undefined
                }
              >
                {!values.departLocation &&
                  <IconButton
                    icon={<CurrentLocationSVG />}
                    onClick={() => {
                      const newValues = {
                        ...values,
                        departLocation: "Current Location",
                      }

                      if (!isSearchMode) {
                        props.onSearchModeChange(true, newValues)
                      } else {
                        props.onValuesChange(newValues)
                      }
                    }}
                  />}
              </BoxField>

              {isSearchMode
                ? <Link
                    to={routes.rides.root(values.mode)}
                    onClick={() => props.onSearchModeChange(false, values)}
                    title="Close"
                  >
                    <IconButton
                      className={styles.closeButton}
                      theme={{ ripple: styles.closeRipple } as any}
                      icon={<CloseSVG className={styles.closeIcon} />}
                    />
                  </Link>
                : <RadioGroup field="mode" className={styles.modeSwitch}>
                    <ModeButton mode="request" />
                    <ModeButton mode="offer" />
                  </RadioGroup>}
            </div>

            <div
              className={classnames(
                styles.headerBottom,
                isSearchMode && styles.inSearchMode
              )}
            >
              <DownChevronSVG className={styles.downChevron} />
              <BoxField field="arriveLocation" placeholder="Destination" />
            </div>

            {isSearchMode && <input type="submit" hidden />}
          </form>}
      </Form>
    </header>
  )
}

export default RideListHeader
