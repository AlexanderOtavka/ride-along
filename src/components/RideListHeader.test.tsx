/**
 * @file RideListHeader.test.tsx
 *
 * Created by Zander Otavka on 6/6/17.
 * Copyright (C) 2016  Grinnell AppDev.
 *
 * @license
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import React from "react"
import { create as reactTestRender } from "react-test-renderer"
import { mount as enzymeMount } from "enzyme"
import { Router, matchPath } from "react-router-dom"
import { createMemoryHistory } from "history"

import RideListHeader from "./RideListHeader"

function getRouteMocks(url: string) {
  const history = createMemoryHistory({ initialEntries: [url] })
  const { location } = history
  const match = matchPath<{ 0: "search" | undefined }>(location.pathname, {
    path: "/(search)?",
    exact: true,
  })

  if (!match) {
    throw TypeError(`Invalid url. Must match path. Got "${url}"`)
  }

  return { history, location, match }
}

function itWhenAtURL(url: string) {
  it(`when at ${url}`, () => {
    const routeMocks = getRouteMocks(url)
    const component = reactTestRender(
      <Router history={routeMocks.history}>
        <RideListHeader {...routeMocks} />
      </Router>
    )

    expect(component.toJSON()).toMatchSnapshot()
  })
}

describe("RideListHandler", () => {
  itWhenAtURL("/")
  itWhenAtURL("/?departlocation=foo&arrivelocation=bar")
  itWhenAtURL("/?mode=request")
  itWhenAtURL("/?mode=offer")
  itWhenAtURL("/search")
  itWhenAtURL("/search?mode=request")
  itWhenAtURL("/search?mode=offer")
  itWhenAtURL("/search?mode=offer&departLocation=foo&arriveLocation=bar")
  itWhenAtURL("/search?mode=request&departLocation=foo")
  itWhenAtURL("/search?mode=offer&arriveLocation=bar")

  it("switches to search when form is submitted", () => {
    const routeMocks = getRouteMocks("/")
    const { history } = routeMocks
    const component = enzymeMount(
      <Router history={history}>
        <RideListHeader {...routeMocks} />
      </Router>
    )

    component.find("form").simulate("submit")
    expect(history.location.pathname).toBe("/search")
    expect(history.location.search).toBe("?mode=request")
    expect(history.length).toBe(2)
  })

  describe("mode switch", () => {
    const routeMocks = getRouteMocks("/")
    const { history } = routeMocks
    const component = enzymeMount(
      <Router history={history}>
        <RideListHeader {...routeMocks} />
      </Router>
    )

    it("starts out with request selected", () => {
      expect(history.location.pathname).toBe("/")
      expect(history.location.search).toBe("?mode=request")
      expect(history.length).toBe(1)
    })

    it("switches to offer mode", () => {
      component.find("ModeButton").last().find("input").simulate("click")
      expect(history.location.pathname).toBe("/")
      expect(history.location.search).toBe("?mode=offer")
      expect(history.length).toBe(1)
    })

    it("stays in offer mode when clicked again", () => {
      component.find("ModeButton").last().find("input").simulate("click")
      expect(history.location.pathname).toBe("/")
      expect(history.location.search).toBe("?mode=offer")
      expect(history.length).toBe(1)
    })

    it("switches back to request mode", () => {
      component.find("ModeButton").first().find("input").simulate("click")
      expect(history.location.pathname).toBe("/")
      expect(history.location.search).toBe("?mode=request")
      expect(history.length).toBe(1)
    })
  })

  describe("search box", () => {
    it("changes the url when depart search box changes", () => {
      const routeMocks = getRouteMocks("/search")
      const { history } = routeMocks
      const component = enzymeMount(
        <Router history={history}>
          <RideListHeader {...routeMocks} />
        </Router>
      )

      component
        .find("BoxField")
        .first()
        .find("input")
        .simulate("change", { target: { value: "foo" } })

      expect(history.location.pathname).toBe("/search")
      expect(history.location.search).toBe("?mode=request&departLocation=foo")
      expect(history.length).toBe(1)
    })

    it("changes the url when arrive search box changes", () => {
      const routeMocks = getRouteMocks("/search")
      const { history } = routeMocks
      const component = enzymeMount(
        <Router history={history}>
          <RideListHeader {...routeMocks} />
        </Router>
      )

      component
        .find("BoxField")
        .last()
        .find("input")
        .simulate("change", { target: { value: "bar" } })

      expect(history.location.pathname).toBe("/search")
      expect(history.location.search).toBe("?mode=request&arriveLocation=bar")
      expect(history.length).toBe(1)
    })
  })
})
