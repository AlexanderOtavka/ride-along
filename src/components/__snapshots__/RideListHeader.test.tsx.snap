// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RideListHandler when at / 1`] = `
<header
  className="header request"
>
  <form
    action="/search"
    method="get"
    onSubmit={[Function]}
  >
    <div
      className="headerTop"
    >
      <div
        className="boxField"
      >
        <input
          autoFocus={false}
          className="input"
          disabled={false}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          type="submit"
          value="Request a ride"
        />
        <div
          className="backdrop"
        />
      </div>
      <div
        className="FormInput"
      >
        <div
          className="FormError"
          style={
            Object {
              "display": "none",
            }
          }
        >
          
        </div>
        <radiogroup
          className="modeSwitch"
        >
          <label
            className="modeButton request"
            title="Request rides"
          >
            <input
              checked={true}
              className="input"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              type="radio"
            />
            <div
              className="button request"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              <svg />
            </div>
          </label>
          <label
            className="modeButton offer"
            title="Offer rides"
          >
            <input
              checked={false}
              className="input"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              type="radio"
            />
            <div
              className="button offer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              <svg />
            </div>
          </label>
        </radiogroup>
      </div>
    </div>
    <div
      className="headerBottom"
    >
      <svg
        className="downChevron"
      />
      <div
        className="boxField"
      >
        <input
          className="input"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          placeholder="Destination"
          value=""
        />
        <div
          className="backdrop"
        />
      </div>
    </div>
  </form>
</header>
`;

exports[`RideListHandler when at /?departlocation=foo&arrivelocation=bar 1`] = `
<header
  className="header request"
>
  <form
    action="/search"
    method="get"
    onSubmit={[Function]}
  >
    <div
      className="headerTop"
    >
      <div
        className="boxField"
      >
        <input
          autoFocus={false}
          className="input"
          disabled={false}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          type="submit"
          value="Request a ride"
        />
        <div
          className="backdrop"
        />
      </div>
      <div
        className="FormInput"
      >
        <div
          className="FormError"
          style={
            Object {
              "display": "none",
            }
          }
        >
          
        </div>
        <radiogroup
          className="modeSwitch"
        >
          <label
            className="modeButton request"
            title="Request rides"
          >
            <input
              checked={true}
              className="input"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              type="radio"
            />
            <div
              className="button request"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              <svg />
            </div>
          </label>
          <label
            className="modeButton offer"
            title="Offer rides"
          >
            <input
              checked={false}
              className="input"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              type="radio"
            />
            <div
              className="button offer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              <svg />
            </div>
          </label>
        </radiogroup>
      </div>
    </div>
    <div
      className="headerBottom"
    >
      <svg
        className="downChevron"
      />
      <div
        className="boxField"
      >
        <input
          className="input"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          placeholder="Destination"
          value=""
        />
        <div
          className="backdrop"
        />
      </div>
    </div>
  </form>
</header>
`;

exports[`RideListHandler when at /?mode=offer 1`] = `
<header
  className="header offer"
>
  <form
    action="/search"
    method="get"
    onSubmit={[Function]}
  >
    <div
      className="headerTop"
    >
      <div
        className="boxField"
      >
        <input
          autoFocus={false}
          className="input"
          disabled={false}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          type="submit"
          value="Offer a ride"
        />
        <div
          className="backdrop"
        />
      </div>
      <div
        className="FormInput"
      >
        <div
          className="FormError"
          style={
            Object {
              "display": "none",
            }
          }
        >
          
        </div>
        <radiogroup
          className="modeSwitch"
        >
          <label
            className="modeButton request"
            title="Request rides"
          >
            <input
              checked={false}
              className="input"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              type="radio"
            />
            <div
              className="button request"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              <svg />
            </div>
          </label>
          <label
            className="modeButton offer"
            title="Offer rides"
          >
            <input
              checked={true}
              className="input"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              type="radio"
            />
            <div
              className="button offer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              <svg />
            </div>
          </label>
        </radiogroup>
      </div>
    </div>
    <div
      className="headerBottom"
    >
      <svg
        className="downChevron"
      />
      <div
        className="boxField"
      >
        <input
          className="input"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          placeholder="Destination"
          value=""
        />
        <div
          className="backdrop"
        />
      </div>
    </div>
  </form>
</header>
`;

exports[`RideListHandler when at /?mode=request 1`] = `
<header
  className="header request"
>
  <form
    action="/search"
    method="get"
    onSubmit={[Function]}
  >
    <div
      className="headerTop"
    >
      <div
        className="boxField"
      >
        <input
          autoFocus={false}
          className="input"
          disabled={false}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          type="submit"
          value="Request a ride"
        />
        <div
          className="backdrop"
        />
      </div>
      <div
        className="FormInput"
      >
        <div
          className="FormError"
          style={
            Object {
              "display": "none",
            }
          }
        >
          
        </div>
        <radiogroup
          className="modeSwitch"
        >
          <label
            className="modeButton request"
            title="Request rides"
          >
            <input
              checked={true}
              className="input"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              type="radio"
            />
            <div
              className="button request"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              <svg />
            </div>
          </label>
          <label
            className="modeButton offer"
            title="Offer rides"
          >
            <input
              checked={false}
              className="input"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              type="radio"
            />
            <div
              className="button offer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              <svg />
            </div>
          </label>
        </radiogroup>
      </div>
    </div>
    <div
      className="headerBottom"
    >
      <svg
        className="downChevron"
      />
      <div
        className="boxField"
      >
        <input
          className="input"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          placeholder="Destination"
          value=""
        />
        <div
          className="backdrop"
        />
      </div>
    </div>
  </form>
</header>
`;

exports[`RideListHandler when at /search 1`] = `
<header
  className="header request"
>
  <form
    action="/search"
    method="get"
    onSubmit={[Function]}
  >
    <div
      className="headerTop"
    >
      <div
        className="boxField"
      >
        <input
          autoFocus={true}
          className="input"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          placeholder="Departure location"
          value=""
        />
        <div
          className="backdrop"
        />
      </div>
      <a
        href="/?mode=undefined"
        onClick={[Function]}
        title="Close"
      >
        <button
          className=" closeButton"
          data-react-toolbox="button"
          disabled={false}
          href={undefined}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          <svg
            className="closeIcon"
          />
        </button>
      </a>
    </div>
    <div
      className="headerBottom inSearchMode"
    >
      <svg
        className="downChevron"
      />
      <div
        className="boxField"
      >
        <input
          className="input"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          placeholder="Destination"
          value=""
        />
        <div
          className="backdrop"
        />
      </div>
    </div>
    <input
      hidden={true}
      type="submit"
    />
  </form>
</header>
`;

exports[`RideListHandler when at /search?mode=offer 1`] = `
<header
  className="header offer"
>
  <form
    action="/search"
    method="get"
    onSubmit={[Function]}
  >
    <div
      className="headerTop"
    >
      <div
        className="boxField"
      >
        <input
          autoFocus={true}
          className="input"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          placeholder="Departure location"
          value=""
        />
        <div
          className="backdrop"
        />
      </div>
      <a
        href="/?mode=offer"
        onClick={[Function]}
        title="Close"
      >
        <button
          className=" closeButton"
          data-react-toolbox="button"
          disabled={false}
          href={undefined}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          <svg
            className="closeIcon"
          />
        </button>
      </a>
    </div>
    <div
      className="headerBottom inSearchMode"
    >
      <svg
        className="downChevron"
      />
      <div
        className="boxField"
      >
        <input
          className="input"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          placeholder="Destination"
          value=""
        />
        <div
          className="backdrop"
        />
      </div>
    </div>
    <input
      hidden={true}
      type="submit"
    />
  </form>
</header>
`;

exports[`RideListHandler when at /search?mode=offer&arriveLocation=bar 1`] = `
<header
  className="header offer"
>
  <form
    action="/search"
    method="get"
    onSubmit={[Function]}
  >
    <div
      className="headerTop"
    >
      <div
        className="boxField"
      >
        <input
          autoFocus={false}
          className="input"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          placeholder="Departure location"
          value=""
        />
        <div
          className="backdrop"
        />
      </div>
      <a
        href="/?mode=offer"
        onClick={[Function]}
        title="Close"
      >
        <button
          className=" closeButton"
          data-react-toolbox="button"
          disabled={false}
          href={undefined}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          <svg
            className="closeIcon"
          />
        </button>
      </a>
    </div>
    <div
      className="headerBottom inSearchMode"
    >
      <svg
        className="downChevron"
      />
      <div
        className="boxField"
      >
        <input
          className="input"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          placeholder="Destination"
          value="bar"
        />
        <div
          className="backdrop"
        />
      </div>
    </div>
    <input
      hidden={true}
      type="submit"
    />
  </form>
</header>
`;

exports[`RideListHandler when at /search?mode=offer&departLocation=foo&arriveLocation=bar 1`] = `
<header
  className="header offer"
>
  <form
    action="/search"
    method="get"
    onSubmit={[Function]}
  >
    <div
      className="headerTop"
    >
      <div
        className="boxField"
      >
        <input
          autoFocus={false}
          className="input"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          placeholder="Departure location"
          value="foo"
        />
        <div
          className="backdrop"
        />
      </div>
      <a
        href="/?mode=offer"
        onClick={[Function]}
        title="Close"
      >
        <button
          className=" closeButton"
          data-react-toolbox="button"
          disabled={false}
          href={undefined}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          <svg
            className="closeIcon"
          />
        </button>
      </a>
    </div>
    <div
      className="headerBottom inSearchMode"
    >
      <svg
        className="downChevron"
      />
      <div
        className="boxField"
      >
        <input
          className="input"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          placeholder="Destination"
          value="bar"
        />
        <div
          className="backdrop"
        />
      </div>
    </div>
    <input
      hidden={true}
      type="submit"
    />
  </form>
</header>
`;

exports[`RideListHandler when at /search?mode=request 1`] = `
<header
  className="header request"
>
  <form
    action="/search"
    method="get"
    onSubmit={[Function]}
  >
    <div
      className="headerTop"
    >
      <div
        className="boxField"
      >
        <input
          autoFocus={true}
          className="input"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          placeholder="Departure location"
          value=""
        />
        <div
          className="backdrop"
        />
      </div>
      <a
        href="/?mode=request"
        onClick={[Function]}
        title="Close"
      >
        <button
          className=" closeButton"
          data-react-toolbox="button"
          disabled={false}
          href={undefined}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          <svg
            className="closeIcon"
          />
        </button>
      </a>
    </div>
    <div
      className="headerBottom inSearchMode"
    >
      <svg
        className="downChevron"
      />
      <div
        className="boxField"
      >
        <input
          className="input"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          placeholder="Destination"
          value=""
        />
        <div
          className="backdrop"
        />
      </div>
    </div>
    <input
      hidden={true}
      type="submit"
    />
  </form>
</header>
`;

exports[`RideListHandler when at /search?mode=request&departLocation=foo 1`] = `
<header
  className="header request"
>
  <form
    action="/search"
    method="get"
    onSubmit={[Function]}
  >
    <div
      className="headerTop"
    >
      <div
        className="boxField"
      >
        <input
          autoFocus={false}
          className="input"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          placeholder="Departure location"
          value="foo"
        />
        <div
          className="backdrop"
        />
      </div>
      <a
        href="/?mode=request"
        onClick={[Function]}
        title="Close"
      >
        <button
          className=" closeButton"
          data-react-toolbox="button"
          disabled={false}
          href={undefined}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          <svg
            className="closeIcon"
          />
        </button>
      </a>
    </div>
    <div
      className="headerBottom inSearchMode"
    >
      <svg
        className="downChevron"
      />
      <div
        className="boxField"
      >
        <input
          className="input"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          placeholder="Destination"
          value=""
        />
        <div
          className="backdrop"
        />
      </div>
    </div>
    <input
      hidden={true}
      type="submit"
    />
  </form>
</header>
`;
